packages/pigeon/lib/dart_generator.dart

```diff
@@ -200,24 +200,38 @@ final BinaryMessenger$nullTag _binaryMessenger;
         final String returnType =
             _makeGenericTypeArguments(func.returnType, nullTag);
         final String castCall = _makeGenericCastCall(func.returnType, nullTag);
+        const String accessor = 'replyMap[\'${Keys.result}\']';
+        final String unwrapper =
+            func.returnType.isNullable ? '' : unwrapOperator;
         final String returnStatement = func.returnType.isVoid
             ? 'return;'
-            : 'return (replyMap[\'${Keys.result}\'] as $returnType$nullTag)$unwrapOperator$castCall;';
+            : 'return ($accessor as $returnType$nullTag)$unwrapper$castCall;';
+        // On iOS we can return nil from functions to accommodate error
+        // handling.  Returning a nil value and not returning an error is an
+        // exception.
```

Comment:
This comment got separated from the code it's about (which is now at line 226).

Reply:
done